modules/Merkle.js

```diff
@@ -10,7 +11,7 @@ class MerkleTree {
         for (let i = 0; i < leaves.length; i += 1) {
             const hash = abi.soliditySHA3(
                 ['bytes32', 'uint256'],
-                [new BN(Buffer.from(leaves[i], 'utf8').toString('hex'), 16), i],
+                [leaves[i], i],
```

Comment:
Just in case leaves[i] looks like '0xabc...' or like some decimal number, better to be sure:
```javascript
[Utilities.normalizehex(Buffer.from(leaves[i], 'utf8').toString('hex')), i],
```
Example why:
```javascript
> EthereumAbi.soliditySHA3(['bytes32'], ['0x' + Buffer.from('0', 'utf-8').toString('hex')]).toString('hex')
'7b1cffef36bc044d4d86b29ca4bdd91e42c6e8193372ca09a109da9b44281e32'
> EthereumAbi.soliditySHA3(['bytes32'], ['0']).toString('hex')
'7b1cffef36bc044d4d86b29ca4bdd91e42c6e8193372ca09a109da9b44281e32'
> EthereumAbi.soliditySHA3(['bytes32'], ['0x' + Buffer.from('0x1', 'utf-8').toString('hex')]).toString('hex')
'535359ca58bd5467016c9606db690cab65ccff86808ee8fd09b14555d0e8b024'
> EthereumAbi.soliditySHA3(['bytes32'], ['0x1']).toString('hex')
'48078cfed56339ea54962e72c37c7f588fc4f8e5bc173827ba75cb10a63a96a5'
```